File: 5.py
class Solution(object):
    def longestPalindrome(self, s):
        """
        :type s: str
        :rtype: str
        """
        if not s: return ""
        res = ""
        for i in range(len(s)):
            for k in range(2):
                temp = self.helper(s, i, i+k)
                if len(res) < len(temp):
                    res = temp
        return res
    def helper(self, s, l, r):
        while l>=0 and r < len(s) and s[l] == s[r]:
            l -= 1
            r += 1
        return s[l+1:r]

File: 5.cpp
class Solution {
public:
    string longestPalindrome(string s) {
        if ( s.size() < 2){
            return s;
        }
        int maxLen = 0;
        int start = 0;
        for(int i = 0; i < s.size() - 1; i++){
            isPalin(s, i, i+1, start, maxLen);
            isPalin(s, i, i, start, maxLen);
        }
        return s.substr(start, maxLen);
    }
    
    void isPalin(string s, int l, int r, int& start, int& maxLen){
        while(l >= 0 and r < s.size() and s[l] == s[r]){
            l--;
            r++;
        }
        if(maxLen < r - l -1){
            maxLen = r- l - 1;
            start = l + 1;
        }
    }
    
    
}; 
